dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/redshift/RedshiftClusterUsageMetricsTask.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.redshift;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatch;
+import com.amazonaws.services.cloudwatch.model.Datapoint;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+import com.amazonaws.services.cloudwatch.model.GetMetricStatisticsRequest;
+import com.amazonaws.services.cloudwatch.model.GetMetricStatisticsResult;
+import com.amazonaws.services.redshift.AmazonRedshift;
+import com.amazonaws.services.redshift.model.Cluster;
+import com.amazonaws.services.redshift.model.DescribeClustersRequest;
+import com.google.auto.value.AutoValue;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.connector.ZonedInterval;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang3.tuple.Pair;
+
+/** Extraction task to get Redshift time series metrics from AWS CloudWatch API. */
+public class RedshiftClusterUsageMetricsTask extends AbstractAwsApiTask {
+
+  protected static enum MetricName {
+    CPUUtilization,
```

Comment:
"CPUUtilization" is the AWS metric name so it has to be like this. I could attach additional value to the Enum, but this seems redundant in this case.

Reply:
In case there is an external API that requires this exact value, I would suggest introducing a String constant. The enum is not required for `MetricName`, because there is no comparison of the enum constants in the code in this PR (like `if` or `switch` over the enum constants). However, enum also works, if the serialized form is passed to the constructor as String.